engine/flink/components/dev-table/src/main/scala/pl/touk/nussknacker/engine/flink/table/aggregate/TableAggregation.scala

```diff
@@ -0,0 +1,129 @@
+package pl.touk.nussknacker.engine.flink.table.aggregate
+
+import org.apache.flink.api.common.RuntimeExecutionMode
+import org.apache.flink.api.common.functions.{FlatMapFunction, RuntimeContext}
+import org.apache.flink.configuration.Configuration
+import org.apache.flink.streaming.api.datastream.DataStream
+import org.apache.flink.streaming.api.functions.ProcessFunction
+import org.apache.flink.table.api.Expressions.{$, call}
+import org.apache.flink.table.api.bridge.java.StreamTableEnvironment
+import org.apache.flink.types.Row
+import org.apache.flink.util.Collector
+import pl.touk.nussknacker.engine.api.VariableConstants.KeyVariableName
+import pl.touk.nussknacker.engine.api
+import pl.touk.nussknacker.engine.api._
+import pl.touk.nussknacker.engine.api.runtimecontext.{ContextIdGenerator, EngineRuntimeContext}
+import pl.touk.nussknacker.engine.flink.api.process.{
+  AbstractLazyParameterInterpreterFunction,
+  FlinkCustomNodeContext,
+  FlinkCustomStreamTransformation
+}
+import pl.touk.nussknacker.engine.flink.table.aggregate.TableAggregation.{
+  aggregateByInternalColumnName,
+  groupByInternalColumnName
+}
+import pl.touk.nussknacker.engine.flink.table.utils.NestedRowConversions.ColumnFlinkSchema
+import pl.touk.nussknacker.engine.flink.table.utils.TypeConversions.getFlinkTypeForNuTypeOrThrow
+import pl.touk.nussknacker.engine.flink.table.utils.{NestedRowConversions, RowConversions}
+
+object TableAggregation {
+  private val aggregateByInternalColumnName = "aggregateByInternalColumn"
+  private val groupByInternalColumnName     = "groupByInternalColumn"
+}
+
+class TableAggregation(
+    groupByLazyParam: LazyParameter[AnyRef],
+    aggregateByLazyParam: LazyParameter[AnyRef],
+    selectedAggregator: Aggregator,
+    nodeId: NodeId
+) extends FlinkCustomStreamTransformation
+    with Serializable {
+
+  override def transform(
+      start: DataStream[Context],
+      context: FlinkCustomNodeContext
+  ): DataStream[ValueWithContext[AnyRef]] = {
+    val env = start.getExecutionEnvironment
+    // Setting batch mode to enable global window operations. If source is unbounded it will throw a runtime exception
+    env.setRuntimeMode(RuntimeExecutionMode.BATCH)
```

Comment:
Uhh. Am I right that it changes the runtime mode into the batch mode for the rest of data stream transformations? We have to talk about that. I though that we can treat it as a technical detail that can be switched on the level of scenario properties. Even if it is impossible now, I think that it should be done somewhere near the `ExecutionConfigPreparer` or at least in source with some TODO

Reply:
Right, I've moved it to the source, parameterized it on level of config of component provider and added todos.